src/commands/registries/azure/deployImageToAci.ts

```diff
@@ -0,0 +1,83 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { IActionContext, parseError } from 'vscode-azureextensionui';
+import { ext } from '../../../extensionVariables';
+import { localize } from '../../../localize';
+import { ContextTreeItem } from '../../../tree/contexts/ContextTreeItem';
+import { registryExpectedContextValues } from '../../../tree/registries/registryContextValues';
+import { RemoteTagTreeItem } from '../../../tree/registries/RemoteTagTreeItem';
+import { executeAsTask } from '../../../utils/executeAsTask';
+import { execAsync } from '../../../utils/spawnAsync';
+import { addImageTaggingTelemetry } from '../../images/tagImage';
+
+export async function deployImageToAci(context: IActionContext, node?: RemoteTagTreeItem): Promise<void> {
+    if (!node) {
+        node = await ext.registriesTree.showTreeItemPicker<RemoteTagTreeItem>([registryExpectedContextValues.dockerHub.tag, registryExpectedContextValues.dockerV2.tag], context);
+    }
+
+    const aciContext = await ext.contextsTree.showTreeItemPicker<ContextTreeItem>([/aciContext;/i], context);
+
+    // Switch to the other context if needed
+    if (!aciContext.current) {
+        await vscode.commands.executeCommand('vscode-docker.contexts.use', aciContext);
+    }
+
+    // Log in to the registry to ensure the run actually succeeds
+    // If a registry was found/chosen and is still the same as the final tag's registry, try logging in
+    await vscode.commands.executeCommand('vscode-docker.registries.logInToDockerCli', node.parent.parent);
+
+    const progressOptions = {
+        location: vscode.ProgressLocation.Notification,
+        title: localize('vscode-docker.commands.registries.deployImageToAci.gettingPorts', 'Determining ports from image...'),
+    };
+    const ports = await vscode.window.withProgress(progressOptions, async () => {
+        return getImagePorts(node.fullTag);
+    });
+    const portsArg = ports.map(port => `-p ${port}:${port}`).join(' ');
+
+    addImageTaggingTelemetry(context, node.fullTag, '');
+
+    const command = `docker --context ${aciContext.name} run -d ${portsArg} ${node.fullTag}`;

```

Comment:
Since we set the chosen ACI context as the current one, is `--context` option necessary here?

Reply:
(See also above). It's not technically necessary but also not harmful. We're doing things in multiple contexts already so I'm inclined to leave it, but I don't feel strongly.

Reply:
I am fine with that